### PR TITLE
[CMake] Check AV_EIGEN_MEMORY_ALIGNMENT when building dependencies

### PR DESCRIPTION
**Pull request by p12tic**
This is follow up after https://github.com/alicevision/AliceVision/pull/1196.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -225,11 +225,29 @@ ExternalProject_Add(${TBB_TARGET}
...
+# in C++17. While we can't upgrade to C++17 just yet, some compilers support overaligned
+# allocation feature with a separate flag, so use it if alignment is enabled in Eigen.
+# See https://eigen.tuxfamily.org/dox/group__TopicUnalignedArrayAssert.html
+if(AV_EIGEN_MEMORY_ALIGNMENT)
+  if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 7.1)
+    set(EIGEN_CMAKE_ALIGNMENT_FLAGS "-DCMAKE_CXX_FLAGS:-faligned-new")
+  elseif(CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 6.0)
+    set(EIGEN_CMAKE_ALIGNMENT_FLAGS "-DCMAKE_CXX_FLAGS:-faligned-new")
+  elseif(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 19.12)
+    set(EIGEN_CMAKE_ALIGNMENT_FLAGS "-DCMAKE_CXX_FLAGS:/Zc:alignedNew")
```
Comment (fabiencastan) [inline CMakeLists.txt:239]:
```suggestion
  if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 7.1)
    set(EIGEN_CMAKE_ALIGNMENT_FLAGS "-DCMAKE_CXX_FLAGS:STRING=-faligned-new")
  elseif(CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 6.0)
    set(EIGEN_CMAKE_ALIGNMENT_FLAGS "-DCMAKE_CXX_FLAGS:STRING=-faligned-new")
  elseif(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 19.12)
    set(EIGEN_CMAKE_ALIGNMENT_FLAGS "-DCMAKE_CXX_FLAGS:STRING=/Zc:alignedNew")
```

Comment (p12tic) [reply CMakeLists.txt:239]: Interesting that oversight this didn't cause a compile error.

Comment (fabiencastan) [reply CMakeLists.txt:239]: yeah, cmake is unfortunately not strict enough for many things...

Comment (p12tic) [reply CMakeLists.txt]: Fixed now. Turns out that I didn't compile locally correctly (used older name for `AV_EIGEN_MEMORY_ALIGNMENT`).

**Review by fabiencastan — APPROVED** [review]

**Comment by fabiencastan** [comment]
@p12tic Could you rebase to fix the conflict?

**Comment by p12tic** [comment]
@fabiencastan Rebased, thanks.

**Review by fabiencastan — APPROVED** [review]

**Pull request merged**